File: ts/src/vector/sequence/sequenceVector.ts
import Vector from "../vector";

export abstract class SequenceVector<T extends ArrayBufferView, K> extends Vector<T, K> {
    protected readonly delta: K;

    protected constructor(name: string, baseValueBuffer: T, delta: K, size: number) {
        super(name, baseValueBuffer, size);
        this.delta = delta;
    }
}

File: java/mlt-core/src/main/java/org/maplibre/mlt/vector/sequence/SequenceVector.java
package org.maplibre.mlt.vector.sequence;

import java.nio.Buffer;
import org.maplibre.mlt.vector.Vector;

public abstract class SequenceVector<T extends Buffer, K> extends Vector<T, K> {
  protected final K delta;

  public SequenceVector(String name, T baseValueBuffer, K delta, int size) {
    super(name, baseValueBuffer, size);
    this.delta = delta;
    this.size = size;
  }
}
